libs/barista-components/button/src/_button-theme.scss

```
@@ -1,227 +1,271 @@
-@import '../../core/src/theming/theming';
-@import '../../core/src/style/interactive-common';
-
-@mixin dt-theme-button($theme) {
-  $palette-names: 'main', 'warning', 'cta';
-  $default-palette: 'main';
-
-  @each $name in $palette-names {
-    $is-dark: map-get($theme, 'is-dark');
-    $palette: dt-get-theme-palette($theme, $name);
-    $default-color: dt-get-theme-color($palette, 'default');
-    $hover-color: dt-get-theme-color($palette, 'hover');
-    $active-color: dt-get-theme-color($palette, 'active');
-
-    // Temporary fix to not make button color depend on theme.
-    // We only have to override the main color because the warning and
-    // cta colors are the same across all themes for now.
-    @if ($name == 'main') {
-      @if ($is-dark == true) {
-        $default-color: $turquoise-500;
-        $hover-color: $turquoise-400;
-        $active-color: $turquoise-300;
-      } @else {
-        $default-color: $turquoise-600;
-        $hover-color: $turquoise-700;
-        $active-color: $turquoise-800;
-      }
-    }
 
-    $namespace: '.dt-button';
-    @if $name != $default-palette {
-      $namespace: '.dt-button.dt-color-#{$name}';
-    }
+.dt-color-warning {
```

Maybe we can scope the color definitions to the host and define a default if they are not set.
For custom properties you can set a default, when a value is not defined. I guess this would be a good fallback if the button is not wrapped into a dtTheme. 

```suggestion
:host {
  --dt-button-default-color: var(--dt-theme-main-default-color, $turquoise500);
  --dt-button-hover-color: var(--dt-theme-main-hover-color, $turquoise600);
  --dt-button-active-color: var(--dt-theme-main-active-color, $turquoise700);
}
:host.dt-color-warning {
  --dt-button-default-color: var(--dt-warning-default-color);
  --dt-button-hover-color: var(--dt-warning-hover-color);
  --dt-button-active-color: var(--dt-warning-active-color);
}
```

Please not that this is not tested code and the turquoise color values may not be correct.

---

Maybe it makes more sense to set the default/fallback on the main.scss file to basically on the :root element. Because if the consumer does not use the dt-theme it will fallback to the root definitions for the custom properties. 
```
:root {
  // i am turquoise by default
--dt-theme-main-
}
```
Otherwise we would end up with default values spread across the components.
What do you think @tomheller @NinaKammerlander

---

I like the idea of defaulting to the turquoise theme in the `main.scss` file with a `:root` selector. 
We probably will still end up with different defaults on a component level, because I think not every default is set to `main` color palette.